src/main/java/org/nuxeo/wopi/lock/LockExpirationListener.java

```diff
@@ -0,0 +1,89 @@
+ */
+package org.nuxeo.wopi.lock;
+
+import static org.nuxeo.wopi.Constants.LOCK_DIRECTORY_DOC_ID;
+import static org.nuxeo.wopi.Constants.LOCK_DIRECTORY_FILE_ID;
+import static org.nuxeo.wopi.Constants.LOCK_DIRECTORY_NAME;
+import static org.nuxeo.wopi.Constants.LOCK_DIRECTORY_SCHEMA_NAME;
+import static org.nuxeo.wopi.Constants.LOCK_EXPIRATION_EVENT;
+
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.core.api.CloseableCoreSession;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.IdRef;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.EventBundle;
+import org.nuxeo.ecm.core.event.PostCommitFilteringEventListener;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * Handles expired WOPI locks by unlocking the related document and removing the stored lock.
+ *
+ * @since 10.3
+ */
+public class LockExpirationListener implements PostCommitFilteringEventListener {
+
+    private static final Log log = LogFactory.getLog(LockExpirationListener.class);
+
+    @Override
+    public void handleEvent(EventBundle eventBundle) {
+        Framework.doPrivileged(() -> {
+            try (Session directorySession = Framework.getService(DirectoryService.class).open(LOCK_DIRECTORY_NAME)) {
+                LockHelper.getExpiredLocksByRepository().entrySet().forEach(entry -> {
+                    String repository = entry.getKey();
+                    List<DocumentModel> lockEntries = entry.getValue();
+                    try (CloseableCoreSession session = CoreInstance.openCoreSession(repository)) {
+                        lockEntries.forEach(lockEntry -> {
+                            handleExpiredLock(session, directorySession, lockEntry, repository);
+                        });
+                    }
+                });
+            }
+        });
+
+    }
+
+    protected void handleExpiredLock(CoreSession session, Session directorySession, DocumentModel entry,
+            String repository) {
+        String docId = (String) entry.getProperty(LOCK_DIRECTORY_SCHEMA_NAME, LOCK_DIRECTORY_DOC_ID);
+        if (log.isDebugEnabled()) {
+            log.debug(String.format(
+                    "WOPI lock expired for document %s/%s, unlocking document and removing lock from directory.",
+                    repository, docId));
+        }
+        // unlock document
+        session.removeLock(new IdRef(docId));
+        // remove WOPI lock from directory
+        directorySession.deleteEntry((String) entry.getProperty(LOCK_DIRECTORY_SCHEMA_NAME, LOCK_DIRECTORY_FILE_ID));
+    }
+
+    @Override
+    public boolean acceptEvent(Event event) {
+        return LOCK_EXPIRATION_EVENT.equals(event.getName());
```

Comment:
You don't need this, this kind of filtering is already done by the `<event>wopiLocksExpiration</event>` in the XML file registering the listener. The listener can be a simple `PostCommitEventListener`.